gotham/src/http/response/mod.rs

```diff
@@ -69,6 +70,102 @@ pub fn create_response(state: &State, status: StatusCode, body: Option<Body>) ->
     res
 }
 
+/// Produces a simple empty `Response` with a `Location` header and a 301
+/// status.
+///
+/// # Examples
+///
+/// ```rust
+/// # extern crate gotham;
+/// # extern crate hyper;
+/// #
+/// # use hyper::{Response, StatusCode};
+/// # use gotham::state::State;
+/// # use gotham::http::response::create_permanent_redirect;
+/// # use gotham::test::TestServer;
+/// # use hyper::header::Location;
+/// fn handler(state: State) -> (State, Response) {
+///     let resp = create_permanent_redirect(&state, "/over-there");
+///
+///     (state, resp)
+/// }
+/// # fn main() {
+/// #     let test_server = TestServer::new(|| Ok(handler)).unwrap();
+/// #     let response = test_server
+/// #         .client()
+/// #         .get("http://example.com/")
+/// #         .perform()
+/// #         .unwrap();
+/// #
+/// #     assert_eq!(response.status(), StatusCode::PermanentRedirect);
+/// #     assert_eq!(
+/// #         response.headers().get::<Location>(),
+/// #         Some(&Location::new("/over-there"))
+/// #     );
+/// # }
+/// ```
+pub fn create_permanent_redirect<L: Into<Cow<'static, str>>>(
+    state: &State,
+    location: L,
+) -> Response {
+    let mut res = Response::new().with_status(StatusCode::PermanentRedirect);
+    {
+        let headers = res.headers_mut();
+        headers.set(XRequestId(request_id(state).into()));
+        headers.set(ContentLength(0));
```

Comment:
AFAIK `ContentLength` isn't valid/required where this is no body content, this could be safely removed.